oada-core/libs/oada-lib-kafka/index.d.ts

```diff
@@ -0,0 +1,24 @@
+// prettier-ignore
+import type { EventEmitter } from 'events'
+
+declare class Base extends EventEmitter {
+}
+
+interface KafkaReguest {
+    msgtype: string
+    code: string
+}
+
+type ConstructorOpts = {
+    consumeTopic: string
+    produceTopic?: string
+    group: string
+    /**
+     * @todo Document these opts
+     */
+    opts?: { [ key:string ]: any }
+}
+export class Responder extends Base {
+    constructor (opts: ConstructorOpts)
+    on (event: 'request', cb: (reg: KafkaReguest) => any): this;
```

Comment:
```suggestion
    on (event: 'request', cb: (reg: KafkaRequest) => any): this;
```

Reply:
In the long run all Kafka messages should have types (and something like `ts-json-schema-generator` to make JSON schema for runtime checking) and this shouldn't be typed as `any` anymore.

Reply:
Sure eventually but I just want websockets to work for now. Though I prefer `typescript-json-schema`.

Reply:
`typescript-json-schema` produced plain wrong schemas so I switched to `ts-json-schema-generator` in trellis-signer which seems to be a more sound implementation.

Reply:
Wrong schemas? It works great for me.

Reply:
Yes, in `trellis-signer` it would not produce the correct schemas under certain conditions. Mostly when there was dependent types. `typescript-json-schema`'s README even mentions that `ts-json-schema-generator` has better support for type alias.

Reply:
I see. that ones seems fine too. I mostly like the support for annotations but that's not needed.

Reply:
@awlayton True. It seems like there is more support going from json schema to typescript (which I guess made sense more descriptive -> less). In terms of message passing, that actually may make more sense.

Reply:
Yeah I suppose we should be defining the schemas in this case no the types.

Reply:
Ideally I think the oada types package would have the schemas of the websoket protocol and then in there, or maybe in another package, we would use a library to compile the schemas into TS types that could be imported.

Reply:
Yes but I think that is out of scope of this PR...